src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -177,6 +177,8 @@ class ImplLookup(private val project: Project, private val items: StdKnownItems)
             result += ty
             ty = if (ty is TyReference) {
                 ty.referenced
+            } else if (ty is TyArray) {
```

Comment:
Maybe convert these `if`s to one `when` construction?

Reply:
Done